### ISSUE TITLE
Clarify jni_x_call macro names

### ISSUE THREAD

**Issue by dmitry-timofeev** [jni-rs/jni-rs#123]
Clarify the naming to express behaviour more precisely:
- [x] `jni_non_null_call` must probably be `jni_non_void_call` — as it *does not* check its return value to be non-null
- [x] `jni_call` must be `jni_non_null_call` — as it works for pointer/references only **and** checks its return value to be non-null.
- [ ] Consider using a separate macro for exception-safe methods.
- [ ] `not_null` → `null_to_err`, see the discussion: https://github.com/jni-rs/jni-rs/pull/150#discussion_r224006732

**Comment by dmitry-timofeev** [comment]
> I see no reason to make these specific macros public.

Yeah, that's rather an implementation detail — and if some JNI methods are missing, it probably makes sense to add them in _this_ library.

**Comment by dmitry-timofeev** [comment]
Updated the description

**Comment by dmitry-timofeev** [comment]
I've mistaken the definition — these macros aren't actually exported :upside_down_face:

**Comment by rib** [comment]
Since this is quite an old, inactive issue now and since I largely refactored the macros in https://github.com/jni-rs/jni-rs/pull/478 since this was opened then it's possible this may effectively be addressed.

Please re-open if you still think there's an issue with the naming / clarify of the internal macros.
